sw/host/opentitanlib/src/app/mod.rs

```diff
@@ -358,6 +369,49 @@ impl TransportWrapperBuilder {
         Ok(())
     }
 
+    fn consolidate_provides_map(
```

Comment:
The uses for this function (and `verify_requires_list`) both take their second argument from `self`.  Can they be rewritten as:
```
fn consolidate_provides_map(&self, provides_map: &mut HashMap<...>) -> Result<()>
```

Reply:
They could, and so could `consolidate_spi_conf_map()` and `consolidate_i2c_conf_map()`.  The last `consolidate_pin_conf_map()` could almost as well, as one of two calls takes the second argument from `self`.
I like the fact that it is obvious looking at the call-sites what data they operate on, though.